app/src/main/kotlin/org/cru/godtools/ui/dashboard/lessons/LessonFilters.kt

```diff
@@ -0,0 +1,140 @@
+import org.cru.godtools.ui.languages.LanguageName
+
+private val DROPDOWN_LESSON_MAX_HEIGHT = 700.dp
+private val DROPDOWN_LESSON_MAX_WIDTH = 350.dp
+
+@Composable
+fun LessonFilters(modifier: Modifier = Modifier) {
+    Column(modifier.fillMaxWidth()) {
+        HorizontalDivider(modifier = Modifier.padding(vertical = 12.dp))
+        Row {
+            Text(
+                stringResource(R.string.dashboard_lessons_section_filter_label),
+                modifier = Modifier
+                    .align(Alignment.CenterVertically)
+                    .padding(end = 36.dp)
+            )
+            LessonLanguageFilter(
+                modifier = Modifier
+                    .weight(1f)
+                    .align(Alignment.CenterVertically)
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+@Composable
+internal fun LessonLanguageFilter(modifier: Modifier = Modifier, viewModel: LessonsViewModel = viewModel()) {
+    val context = LocalContext.current
+    val selectedLanguage by viewModel.selectedLanguage.collectAsState()
+
+    val lessonLanguages by viewModel.filteredLanguages.collectAsState(emptyList())
+    var expanded by remember { mutableStateOf(false) }
+    var query by remember { mutableStateOf("") }
+    val scope = rememberCoroutineScope()
+
+    ElevatedButton(
+        onClick = { expanded = true },
+        modifier = modifier.semantics { role = Role.DropdownList }
+    ) {
+        Text(
+            selectedLanguage.getDisplayName(context, LocalAppLanguage.current),
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            modifier = Modifier.weight(1f),
+        )
+        ExposedDropdownMenuDefaults.TrailingIcon(expanded = expanded)
+        LazyDropdownMenu(
+            expanded = expanded,
+            onDismissRequest = { expanded = false },
+            modifier = Modifier.sizeIn(maxHeight = DROPDOWN_LESSON_MAX_HEIGHT, maxWidth = DROPDOWN_LESSON_MAX_WIDTH)
+        ) {
+            stickyHeader {
+                Surface(color = MaterialTheme.colorScheme.surface) {
+                    SearchBar(
+                        query = query,
+                        onQueryChange = { change ->
+                            scope.launch {
+                                query = change
+                                viewModel.updateQuery(change)
+                            }
+                        },
+                        onSearch = {},
+                        active = false,
+                        onActiveChange = {},
+                        colors = GodToolsTheme.searchBarColors,
+                        placeholder = {
+                            Text(stringResource(R.string.language_settings_downloadable_languages_search))
+                        },
+                        content = {},
+                        modifier = Modifier
+                            .padding(horizontal = 16.dp, vertical = 16.dp)
```

Comment:
I feel like this padding is wrong, that would put a lot of additional vertical whitespace around the search bar in a UI that should probably be more compact